nncf/quantization/algorithms/min_max/algorithm.py

```diff
@@ -359,26 +369,97 @@ def _get_quantization_target_points(self, model: TModel) -> OrderedDict[TargetPo
+                                          nncf_graph: NNCFGraph) -> Optional[List[TargetPoint]]:
+        """
+        Returns target points connected to a first visited node, which are included in quantization_points.
+        A traversal of nncf_graph is started from starting_node.
+
+        :param quantization_points: Quantization target points.
+        :param starting_node: Node from which traversing is started.
+        :param nncf_graph: Instance of NNCFGraph to traverse.
+        :return: First visited target points.
+        """
+        target_node_names_to_qp = {}
+        for q_p in quantization_points:
+            if q_p.target_node_name not in target_node_names_to_qp:
+                target_node_names_to_qp[q_p.target_node_name] = [q_p]
+            else:
+                target_node_names_to_qp[q_p.target_node_name].append(q_p)
+        queue = collections.deque(nncf_graph.get_next_nodes(starting_node))
+        while queue:
+            node = queue.popleft()
+            node_name = node.node_name
+            if node_name in target_node_names_to_qp:
+                return target_node_names_to_qp[node_name]
+            queue.extend(nncf_graph.get_next_nodes(node))
+        return None
+
+    def _get_quantization_points_overflow_fix(self, overflow_fix: OverflowFix,
+                                              quantization_target_points: OrderedDict[TargetPoint, QuantizerConfig],
+                                              nncf_graph: NNCFGraph) -> Set[TargetPoint]:
+        """
+        Returns quantization target points, for whom overflow_fix should be applied.
+        If overflow_fix is enable, returns all weights quantization target points.
+        If overflow_fix is disable, returns an empty set.
+        If overflow_fix is first_layer_only, returns all first weights quantization target points.
+
+        :param overflow_fix: OverflowFix parameter.
+        :param quantization_target_points: Quantization target points.
+        :param nncf_graph: Instance of NNCFGraph to traverse.
+        :return: If overflow_fix is enable, returns all weights quantization target points.
+                 If overflow_fix is disable, returns an empty set.
+                 If overflow_fix is first_layer_only, returns all first weights quantization target points.
+        """
+        weight_quantization_points = set(filter(lambda point: point.is_weight_target_point(),
+                                                quantization_target_points.keys()))
+        if overflow_fix == OverflowFix.ENABLE:
+            return weight_quantization_points
+        if overflow_fix == OverflowFix.FIRST_LAYER:
+            output = set()
+            for input_node in nncf_graph.get_input_nodes():
+                nodes = self._get_first_quantized_weight_nodes(weight_quantization_points, input_node, nncf_graph)
+                if nodes:
+                    output.update(nodes)
```

Comment:
Can `self._get_first_quantized_weight_nodes` always returns `List`? We can simplify code than
```suggestion
               output.extend(self._get_first_quantized_weight_nodes(weight_quantization_points, input_node, nncf_graph))

```

Reply:
Indeed. Updated it